src/test/db.rs

```diff
@@ -318,3 +323,61 @@ async fn db_aggregate_disk_use() {
         .await
         .expect("aggregate with disk use should succeed");
 }
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn create_index_options_defaults() {
+    let defaults = IndexOptionDefaults {
+        storage_engine: doc! { "wiredTiger": doc! {} },
+    };
+    index_option_defaults_test(Some(defaults)).await;
+}
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn create_index_options_defaults_not_specified() {
+    index_option_defaults_test(None).await;
+}
+
+#[function_name::named]
+async fn index_option_defaults_test(defaults: Option<IndexOptionDefaults>) {
+    let _guard = LOCK.run_concurrently().await;
+
+    let mut options = CLIENT_OPTIONS.clone();
+    if TestClient::new().await.is_sharded() {
```

Comment:
These tests originally failed on sharded clusters, so I stripped out the extra hosts similarly to retryable reads.

Reply:
What were the failures you were seeing? I don't think having multiple mongoses should cause problems here but I could be wrong.

Reply:
[This run](https://evergreen.mongodb.com/version/5ed588f057e85a11b81f1688) failed on three sharded clusters when doing `db.createCollection()`, all with namespace not found. Stripping out the extra hosts seemed to do away with the problem, although I didn't investigate much further.

Reply:
So it looks like the error was happening in that patch because you were re-using the same database without calling [`flushRouterConfig`](https://docs.mongodb.com/manual/reference/command/dropDatabase/#replica-set-and-sharded-clusters) post-drop. Your latest patches have different databases per test, so they should work fine even using both mongoses.

Reply:
Gotcha, I'll revert this change.